boards/arm/mps2_an521/board.cmake

```diff
@@ -11,3 +11,10 @@ set(QEMU_FLAGS_${ARCH}
   -vga none
   )
 board_set_debugger_ifnset(qemu)
+
+if (CONFIG_BUILD_WITH_TFM)
+  # Override the binary used by qemu, to use the combined
+  # TF-M (Secure) & Zephyr (Non Secure) image (when running
+  # in-tree tests).
+  set(QEMU_KERNEL_OPTION "-device;loader,file=${CMAKE_BINARY_DIR}/tfm_merged.hex")
```

Comment:
@tejlmand FYI

Reply:
Any implications to this getting set in `samples/tfm_integration/tfm_ipc/CMakeLists.txt` and `samples/tfm_integration/psa_level_1/CMakeLists.txt`?

Reply:
I guess this should be removed from the other places now.

Reply:
> I guess this should be removed from the other places now.

Well, sure. I kept it there, for the samples to be self-sustained, i.e. have the qemu kernel option set, independent of the board setting. But yeap, it can be removed.

Reply:
@galak that's the point, actually. I would like all the samples and tests to run with tfm, so this setting needs to be generic...

Reply:
@ioannisg not directly related to this PR, but I notice we have both:
https://github.com/zephyrproject-rtos/zephyr/blob/8697935882a8daffe63904464ace70a3f4d77200/modules/trusted-firmware-m/CMakeLists.txt#L322-L331

and:
https://github.com/zephyrproject-rtos/zephyr/blob/8697935882a8daffe63904464ace70a3f4d77200/CMakeLists.txt#L1389-L1402

Could you create a task on aligning those, including making the `merged.hex` / `tfm_merged.hex` name configurable like `KERNEL_BIN_NAME` is:
https://github.com/zephyrproject-rtos/zephyr/blob/8697935882a8daffe63904464ace70a3f4d77200/Kconfig.zephyr#L274-L278